x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/perform_rule_installation/perform_rule_installation_handler.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformError } from '@kbn/securitysolution-es-utils';
+import type { KibanaRequest, KibanaResponseFactory } from '@kbn/core/server';
+import { SkipRuleInstallReason } from '../../../../../../common/api/detection_engine/prebuilt_rules';
+import type {
+  PerformRuleInstallationResponseBody,
+  SkippedRuleInstall,
+  PerformRuleInstallationRequestBody,
+} from '../../../../../../common/api/detection_engine/prebuilt_rules';
+import type { SecuritySolutionRequestHandlerContext } from '../../../../../types';
+import { buildSiemResponse } from '../../../routes/utils';
+import { aggregatePrebuiltRuleErrors } from '../../logic/aggregate_prebuilt_rule_errors';
+import { ensureLatestRulesPackageInstalled } from '../../logic/ensure_latest_rules_package_installed';
+import { createPrebuiltRuleAssetsClient } from '../../logic/rule_assets/prebuilt_rule_assets_client';
+import { createPrebuiltRules } from '../../logic/rule_objects/create_prebuilt_rules';
+import { createPrebuiltRuleObjectsClient } from '../../logic/rule_objects/prebuilt_rule_objects_client';
+import { performTimelinesInstallation } from '../../logic/perform_timelines_installation';
+import type { RuleSignatureId, RuleVersion } from '../../../../../../common/api/detection_engine';
+
+export const performRuleInstallationHandler = async (
+  context: SecuritySolutionRequestHandlerContext,
+  request: KibanaRequest<unknown, unknown, PerformRuleInstallationRequestBody>,
+  response: KibanaResponseFactory
+) => {
```

Comment:
nit: Specifying result type would be hight appreciated to reason on what's expected result. TS will help to make sure returned data matches expectations.

Reply:
Yes, that would help. The only problem is that this method can return errors along with a response, so the type there would be a bit bulky. I think the best way is to refactor our route handlers a bit and extract error handler and request wrapping into separate wrappers to leave handler implementations clean. This is related to [your other comment](https://github.com/elastic/kibana/pull/214441#discussion_r1995234279). We can find some time to clean up route implementations, maybe during On week, to address this.